### PR TITLE
2018 03 10 tar perm roles p1 t2

### PR DESCRIPTION
**Pull request by ridz1208**
This pull request `[DESCRIBE WHAT THIS PULL REQUEST IS TRYING TO ACCOMPLISH HERE]`. It `[DESCRIBE HOW IT ACCOMPLISHES THE GOAL HERE]`.

See also: `[PROVIDE REFERENCES TO OTHER TICKETS, PULL REQUESTS, OR MAILING LIST THREADS HERE. OPTIONAL.]`

### CODE REVIEW

**Review by xlecours — CHANGES_REQUESTED** [review]
A few things but I think we should look at a bigger picture here. I thought the permission controler would add/remove permission(s) to a user or role. Doign so it would expect a User or a Role and an array of permission to deal with. Maybe we should talk face to face.

**Comment by ridz1208** [comment]
@xlecours yes it's right it should be and they will be added below that whitespace gap. lets talk when you want

File: php/libraries/Permission.class.inc [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+     *
+     * @param $ID int Identifier
+     *
+     * @throws LorisException when the field is already set to avoid user altering
+     *                        its value and trying to update or deleting it.
+     *
+     * @return void
+     */
+    function setID(int $ID) {
+        // Limit accessibility to this value.
```
[the remark this answers is not in the document]
Comment (ridz1208) [reply php/libraries/Permission.class.inc]: immutability seems very complex to me in this cintext I think we can sit Dave, Melanie, you and I and whoever else is intersted and discuss it in this context but for now the most critical element is the ID

File: php/libraries/PermissionController.class.inc [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+     * @access private
+     */
+    private $permissionDAO;
+
+    /**
+     * Permission constructor.
+     *
+     * @param Database $database database
+     */
+    function __construct(Database $database)
```
[the remark this answers is not in the document]
Comment (ridz1208) [reply php/libraries/PermissionController.class.inc]: sure

**Pull request closed without merging**